### PR TITLE
APIS-6910 - Update lib and change to instant where necessary

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: acceptance/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala [hunk 1]
```diff
@@ -159,13 +159,13 @@ trait ThirdPartyApplicationStub extends WireMockExtensions with ApplicationWithS
     stubFor(get(urlEqualTo(s"/gatekeeper/application/${applicationId.value.toString()}")).willReturn(aResponse().withBody(defaultApplicationWithHistory.toJsonString).withStatus(OK)))
   }
 
-  def stubApplicationForDeleteSuccess(applicationId: ApplicationId) = {
-    stubFor(patch(urlEqualTo(s"/application/${applicationId.value.toString()}")).willReturn(aResponse().withStatus(NO_CONTENT)))
-  }
+  // def stubApplicationForDeleteSuccess(applicationId: ApplicationId) = {
```
Comment (johnsgp) [inline acceptance/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala:162]: Do you want to keep these comments?

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala [hunk 2]
```diff
@@ -679,42 +679,50 @@ class ApplicationServiceSpec extends AsyncHmrcSpec with ResetMocksAfterEachTest
...
+          aRequestor shouldBe emailAddress
+      }
     }
 
-    "propagate ApplicationDeleteFailureResult from connector" in new Setup {
-      val emailAddress  = "email@example.com"
-      val application   = stdApp1.copy(deployedTo = Environment.SANDBOX)
-      val reasons       = "Application deleted by Gatekeeper user"
-      val requestCaptor = ArgCaptor[DeleteApplicationByGatekeeper]
+    /** updateGrantLengthCommandWillSucceed
```
Comment (johnsgp) [inline test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:700]: Is this still work-in-progress?

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala [hunk 3]
```diff
@@ -679,42 +679,50 @@ class ApplicationServiceSpec extends AsyncHmrcSpec with ResetMocksAfterEachTest
...
+      *
+      * val result = await(underTest.updateGrantLength(stdApp1, GrantLength.THREE_MONTHS, gatekeeperUserId)) result shouldBe ApplicationUpdateSuccessResult
+      *
+      * inside(CommandConnectorMock.IssueCommand.verifyCommand(stdApp1.id)) { case ApplicationCommands.ChangeGrantLength(aUser, _, length) => aUser shouldBe gatekeeperUserId length
+      * shouldBe GrantLength.THREE_MONTHS } }
+      */
 
-      ApplicationConnectorMock.Sandbox.DeleteApplication.fails()
+    //   "propagate ApplicationDeleteFailureResult from connector" in new Setup {
```
Comment (johnsgp) [inline test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:708]: More commented out code

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala [hunk 4]
```diff
@@ -679,42 +679,50 @@ class ApplicationServiceSpec extends AsyncHmrcSpec with ResetMocksAfterEachTest
...
-      result shouldBe ApplicationUpdateFailureResult
+    //     val result = await(underTest.deleteApplication(application, gatekeeperUserId, emailAddress))
 
-      verify(underTest).applicationConnectorFor(application)
-      verify(mockSandboxApplicationConnector).deleteApplication(eqTo(application.id), requestCaptor)(*)
-      requestCaptor.value.gatekeeperUser shouldBe gatekeeperUserId
-      requestCaptor.value.requestedByEmailAddress shouldBe emailAddress
-      requestCaptor.value.reasons shouldBe reasons
-    }
+    //     result shouldBe ApplicationUpdateFailureResult
```
Comment (johnsgp) [inline test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:718]: More commented out code

File: test/uk/gov/hmrc/gatekeeper/connectors/ApplicationConnectorSpec.scala [hunk 5]
```diff
@@ -452,36 +453,6 @@ class ApplicationConnectorSpec
     }
   }
 
-  "unsubscribeFromApi" should {
```
Comment (johnsgp) [inline test/uk/gov/hmrc/gatekeeper/connectors/ApplicationConnectorSpec.scala:455]: What is replacing this test? Has coverage dropped?

File: acceptance/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply acceptance/uk/gov/hmrc/gatekeeper/stubs/ThirdPartyApplicationStub.scala:162]: will remove

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:700]: will remove

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:708]: will remove

File: test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply test/uk/gov/hmrc/gatekeeper/services/ApplicationServiceSpec.scala:718]: will update.

**Pull request merged**